zmon_worker_monitor/builtins/plugins/cassandra_wrapper.py

```diff
@@ -54,20 +54,24 @@ def __init__(self, node, keyspace, username=None, password=None, port=9042, conn
         self.connect_timeout = connect_timeout
         self.protocol_version = protocol_version
 
-    def execute(self, stmt):
         auth_provider = None
         if self.__username and self.__password:
             auth_provider = PlainTextAuthProvider(username=self.__username, password=self.__password)
 
         cl = Cluster(self.seeds, connect_timeout=self.connect_timeout, auth_provider=auth_provider,
                      protocol_version=self.protocol_version, port=self.port)
 
-        session = None
-        try:
-            session = cl.connect()
-            session.set_keyspace(self.keyspace)
+        self._session = cl.connect()
+        self._cluster = cl
+        self._session.set_keyspace(self.keyspace)
+
+    def __del__(self):
+        if self._cluster:
```

Comment:
Not needed. Combined with removing the ``except`` in ``execute``, this will always just terminate the connection

Reply:
On a second thought, it makes sense to keep it since we could fail in the ``__init__()`` connect call 😄